### PR TITLE
add optional CPU limit

### PR DESCRIPTION
**Pull request by twaldrop**
Add option to limit available CPUs via grub config

### CODE REVIEW

File: roles/common/tasks/kernel-tuning.yml [hunk 1]
```diff
@@ -28,6 +28,13 @@
               line="GRUB_CMDLINE_LINUX_DEFAULT=\"\""
   notify: update grub config
 
+- name: "Limit available cores via grub"
+  lineinfile: dest=/etc/default/grub
+              regexp="^GRUB_CMDLINE_LINUX_DEFAULT="
+              line="GRUB_CMDLINE_LINUX_DEFAULT=\"{{ cpulimit }}\""
```
Comment (j2sol) [inline roles/common/tasks/kernel-tuning.yml]: This should be `maxcpu={{ cpulimit }}`. You were missing the maxcpu part.

However, this should go into the `GRUB_CMDLINE_LINUX` line, not this one.

Comment (j2sol) [inline roles/common/tasks/kernel-tuning.yml]: This should just be added to the set of boot parameters on the task `Kernel boot parameters:`

To sort out how to add this conditionally, you'll want to add this to the `line=` at the end:

`{{- ' maxcpu=%s' % maxcpu if maxcpu is defined else '' }}`

The spacing is important, as is the `-` after the opening `{{`.

**Comment by twaldrop** [comment]
Updated PR with J2sol's feedback

File: roles/common/tasks/kernel-tuning.yml (hunk 1 above)
Comment (dlundquist) [inline roles/common/tasks/kernel-tuning.yml]: This needs to be added to GRUB_CMDLINE_LINUX below instead.

**Comment by twaldrop** [comment]
Had an issue with my previous amend. PR is updated now.

File: roles/common/tasks/kernel-tuning.yml [hunk 2]
```diff
@@ -40,7 +40,7 @@
 - name: "Kernel boot parameters: disable console screen blanking, enabled serial console on IPMI serial over LAN"
```
Comment (blueboxjesse) [inline roles/common/tasks/kernel-tuning.yml]: Do we want to amend this name to reference the cpu limit function?

**Comment by twaldrop** [comment]
Updated.  Thanks for pointing that out, JP

File: roles/common/tasks/kernel-tuning.yml [hunk 3]
```diff
@@ -37,10 +37,10 @@
 #                   See https://www.kernel.org/doc/Documentation/serial-console.txt
 #  biosdevname=0    Disable naming ethernet devices by bios names
 #
-- name: "Kernel boot parameters: disable console screen blanking, enabled serial console on IPMI serial over LAN"
+- name: "Kernel boot parameters: disable console screen blanking, enabled serial console on IPMI serial over LAN, optional maxcpu"
   lineinfile: dest=/etc/default/grub
               regexp="^GRUB_CMDLINE_LINUX="
-              line="GRUB_CMDLINE_LINUX=\"consoleblank=0 crashkernel=256M nmi_watchdog=1 {{ serial_console_cmdline|default('') }} console=tty0 biosdevname=0\""
+              line="GRUB_CMDLINE_LINUX=\"consoleblank=0 crashkernel=256M nmi_watchdog=1 {{ serial_console_cmdline|default('') }} console=tty0 biosdevname=0\" {{- ' maxcpu=%s' % cpulimit if cpulimit is defined else '' }} "
```
Comment (j2sol) [inline roles/common/tasks/kernel-tuning.yml]: The added bits need to be inside the quote. You've added them outside the quote.

**Comment by j2sol** [comment]
Good to merge when tests pass. :+1:

**Comment by twaldrop** [comment]
test was cancelled in CI a few days ago.  retesting

**Pull request merged**